test/test_action_builder/test_action_builder.py

```diff
@@ -0,0 +1,345 @@
+import unittest
+import requests_mock
+import json
+from parsons import Table, ActionBuilder
+from test.utils import assert_matching_tables
+
+
+class TestActionBuilder(unittest.TestCase):
+    @requests_mock.Mocker()
+    def setUp(self, m):
+        self.subdomain = "fake_subdomain"
+        self.campaign = "fake-campaign"
+        self.api_url = "https://{}.actionbuilder.org/api/rest/v1/campaigns/{}".format(
+            self.subdomain, self.campaign
+        )
+        self.api_key = "fake_key"
+
+        self.bldr = ActionBuilder(
+            api_token=self.api_key, subdomain=self.subdomain, campaign=self.campaign
+        )
+
+        self.fake_datetime = "2023-05-19T00:00:00.000Z"
+        self.fake_date = "2023-05-19"
+
+        self.fake_tag_1 = "Fake Tag 1"
+        self.fake_tag_2 = "Fake Tag 2"
+        self.fake_tag_3 = "Fake Tag 3"
+        self.fake_tag_4 = "Fake Tag 3"
+
+        self.fake_field_1 = "Fake Field 1"
+        self.fake_section = "Fake Section 1"
+
+        self.fake_tags_list_1 = {
```

Comment:
this setup gets pretty involved. you might consider some generator functions that accept args and spit out these more complicated objects

Reply:
You're probably right, but I'm hoping it's ok to save that for the next version of updates to this connector.

Reply:
It's worth thinking about creating some general guidance for implementing factory functions to create test data (and maybe even make some default factory functions that create common objects that people can subclass/build off of). Right now most connectors rely on creating json or python dicts, in the tests like Yotam's done here or imported from separate files, and either way it's messy and tedious to write and generally not ideal. 

But factory methods are a relatively advanced python concept so we haven't wanted to say "just do this" without giving contributors support. If we can make some good guidance/templates and some default classes, we could slowly transition over connectors (especially highly used connectors, and new connectors) to use factory functions and that would probably be better for the project/community overall.